File: Android/AndroidStudioProjects/TesJava/app/src/main/java/com/samzuhalsetiawan/tesjava/SecondActivity.java
package com.samzuhalsetiawan.tesjava;

import androidx.activity.result.ActivityResult;
import androidx.activity.result.ActivityResultCallback;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        Button mButton = findViewById(R.id.btnSendResult);
        mButton.setOnClickListener(v -> {
            Intent intent = new Intent();
            intent.putExtra("EXTRA_DATA", "Hallo Dunia");
            setResult(RESULT_OK, intent);
            finish();
        });

    }
}

File: Android/AndroidStudioProjects/TesKotlin/app/src/main/java/com/samzuhalsetiawan/teskotlin/SecondActivity.kt
package com.samzuhalsetiawan.teskotlin

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import androidx.activity.result.contract.ActivityResultContracts

class SecondActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        findViewById<Button>(R.id.btnSendResult).setOnClickListener {
            val intent = Intent().apply {
                putExtra("EXTRA_DATA", "Hello Dunia")
            }
            setResult(RESULT_OK, intent)
            finish()
        }

    }

}